neo/rawio/intanbinaryrawio.py

```diff
@@ -0,0 +1,438 @@
+    rhd_global_header_base,
+    rhd_global_header_part1,
+    rhd_global_header_v11,
+    rhd_global_header_v13,
+    rhd_global_header_v20,
+    rhd_global_header_final,
+    rhd_signal_group_header,
+    rhd_signal_channel_header,
+    stream_type_to_name,
+)
+
+
+class IntanBinaryRawIO(BaseRawIO):
+    """
+    Class for processing Intan Data when saved in binary format. Requires an
+    `info.rhd` as well one file per signal stream or one file per channel
+
+    Parameters
+    ----------
+    dirname: str, Path
+        The root directory containing the info.rhd file
+
+    """
+
+    extensions = ["rhd", "dat"]
+    rawmode = "one-dir"
+
+    def __init__(self, dirname=""):
+        BaseRawIO.__init__(self)
+        self.dirname = dirname
+
+    def _source_name(self):
+        return self.dirname
+
+    def _parse_header(self):
+
+        dir_path = Path(self.dirname)
+        assert dir_path.is_dir(), (
+            "IntanBinaryRawIO requires the root " "directory containing info.rhd"
+        )
+
+        header_file = dir_path / "info.rhd"
+
+        for file in possible_raw_files:
+            if (dir_path / file).is_file():
+                one_file_per_signal = True
+                break
+            else:
+                one_file_per_signal = False
+
+        self.one_file_per_signal = one_file_per_signal
+
+        if one_file_per_signal:
+            raw_file_dict = create_one_file_per_signal_dict(dir_path)
+        else:
+            raw_file_dict = create_one_file_per_channel_dict(dir_path)
+
+        (
+            self._global_info,
+            self._ordered_channels,
+            data_dtype,
+            self._block_size,
+        ) = read_rhd(header_file)
+
+        self._raw_data = {}
+        for stream_index, sub_datatype in data_dtype.items():
+            if one_file_per_signal:
+                self._raw_data[stream_index] = np.memmap(
+                    raw_file_dict[stream_index], dtype=sub_datatype, mode="r"
+                )
+            else:
+                self._raw_data[stream_index] = []
+                for channel_index, datatype in enumerate(sub_datatype):
+                    self._raw_data[stream_index].append(
+                        np.memmap(
+                            raw_file_dict[stream_index][channel_index],
+                            dtype=[datatype],
+                            mode="r",
+                        )
+                    )
+
+        # check timestamp continuity
+        if one_file_per_signal:
+            timestamp = self._raw_data[6]["timestamp"].flatten()
+        else:
+            timestamp = self._raw_data[6][0]["timestamp"].flatten()
+        assert np.all(np.diff(timestamp) == 1), "timestamp have gaps"
+
+        # signals
+        signal_channels = []
+        for c, chan_info in enumerate(self._ordered_channels):
+            name = chan_info["native_channel_name"]
+            chan_id = str(c)  # the chan_id have no meaning in intan
```

Comment:
Why would be wrong with using `native_channel_name` as a channel_id if the latter has no meaning? or what do you mean by channel id not having a meaning?

Reply:
Again this is a holdover from Sam's RawIO implementation. I kept it the same. Neo has three ways to access channels. `channel_indexes` which are just the columns of the data matrix, `channel_names` which are the names given to the channels (so in this case the 'native_channel_name'), and the `channel_ids` which are something I don't fully understand, but they are typically numeric, but encoded as strings. @samuelgarcia would have to explain when we would use the `channel_ids`.